tower-http/src/add_extension.rs

```diff
@@ -1,14 +1,22 @@
+//! Middleware for adding some shareable value to [request extensions].
+//!
+//! [request extensions]: https://docs.rs/http/latest/http/struct.Extensions.html
+
 use http::Request;
 use std::task::{Context, Poll};
 use tower_layer::Layer;
 use tower_service::Service;
 
+/// [`Layer`] for adding some shareable value to [request extensions].
```

Comment:
since `Layer` is defined in another crate, I *think* this needs a link reference.

Reply:
Nope it seems to be working.

Reply:
Honestly not sure why. There are other times where it doesn't work.